pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -527,6 +542,14 @@ func filterChainMatch(fc *xdslistener.FilterChain, cp *model.EnvoyFilterConfigPa
 			return false
 		}
 	}
+
+	// check match for destination port within the FilterChainMatch
+	if cMatch.PortNumber > 0 &&
+		fc.FilterChainMatch != nil && fc.FilterChainMatch.DestinationPort != nil &&
```

Comment:
nit: proto generated libraries add `.GetX` which will allow chaining of `nil` fields like `fc.GetFilterChainMatch().GetDestinationPort().GetValue()`